src/transactions/test/InvokeHostFunctionTests.cpp

```diff
@@ -40,294 +40,169 @@
 
-        {
-            // submit operation
-            auto tx = sorobanTransactionFrameFromOps(
-                app->getNetworkID(), a1, {transfer}, {}, resources, 100,
-                DEFAULT_TEST_RESOURCE_FEE);
-
-            LedgerTxn ltx(app->getLedgerTxnRoot());
-            TransactionMetaFrame txm(ltx.loadHeader().current().ledgerVersion);
-            REQUIRE(tx->checkValid(*app, ltx, 0, 0, 0));
-            REQUIRE(tx->apply(*app, ltx, txm));
-            ltx.commit();
-        }
-        REQUIRE(preTransferA1Balance - 10 == a1.getBalance());
+    auto accAddr = makeAccountAddress(acc.getPublicKey());
 
-        // Make sure sponsorship info hasn't changed
-        {
-            LedgerTxn ltx(app->getLedgerTxnRoot());
-            checkSponsorship(ltx, a1.getPublicKey(), 1, &root.getPublicKey(), 0,
-                             2, 0, 2);
-            checkSponsorship(ltx, root.getPublicKey(), 0, nullptr, 0, 2, 2, 0);
-        }
-    }
+    // Transfer and mint to account
+    test.transfer(root, accAddr, 10, true);
+    test.mint(issuer, accAddr, 10, true);
 
-    SECTION("trustline")
-    {
-        auto const minBalance = app->getLedgerManager().getLastMinBalance(2);
-        auto issuer = root.create("issuer", minBalance);
-        auto acc = root.create("acc", minBalance);
-        auto sponsor = root.create("sponsor", minBalance * 2);
+    // Now mint by transfering from issuer
+    test.transfer(issuer, accAddr, 10, true);
 
-        Asset idr = makeAsset(issuer, "IDR");
-        root.changeTrust(idr, 100);
-        {
-            auto tx = transactionFrameFromOps(
-                app->getNetworkID(), sponsor,
-                {sponsor.op(beginSponsoringFutureReserves(acc)),
-                 acc.op(changeTrust(idr, 100)),
-                 acc.op(endSponsoringFutureReserves())},
-                {acc});
-
-            LedgerTxn ltx(app->getLedgerTxnRoot());
-            TransactionMetaFrame txm(ltx.loadHeader().current().ledgerVersion);
-            REQUIRE(tx->checkValid(*app, ltx, 0, 0, 0));
-            REQUIRE(tx->apply(*app, ltx, txm));
-            REQUIRE(tx->getResultCode() == txSUCCESS);
-            ltx.commit();
-        }
+    // Now burn by transfering to the issuer and by using burn function
+    test.transfer(acc, makeAccountAddress(issuer.getPublicKey()), 10, true);
+    test.burn(acc, 10, true);
 
-        {
-            LedgerTxn ltx(app->getLedgerTxnRoot());
-            auto tlAsset = assetToTrustLineAsset(idr);
-            checkSponsorship(ltx, trustlineKey(acc, tlAsset), 1,
-                             &sponsor.getPublicKey());
-            checkSponsorship(ltx, acc, 0, nullptr, 1, 2, 0, 1);
-            checkSponsorship(ltx, sponsor, 0, nullptr, 0, 2, 1, 0);
-        }
+    // Now transfer and mint to contractAddress
+    auto contractAddr = makeContractAddress(sha256("contract"));
+    test.transfer(root, contractAddr, 10, true);
+    test.mint(issuer, contractAddr, 10, true);
```

Comment:
We should also try minting/transferring/burning to/from issuer.

Reply:
Ah, I haven't noticed, thanks. But I guess there are still missing cases:
- `burn` from issuer
- `mint` from issuer to issuer

Reply:
Ah that's a good point... I haven't written the test yet, but what I expect to happen is no balance change, but the event will be emitted which is probably and issue.

Reply:
I think the solution is to fail in the host if burn mint or clawback id called on the issuer.

Reply:
https://github.com/stellar/rs-soroban-env/pull/1215